### DISCUSSION TITLE
Help documenting other variants of NVM (Node Version Manager)

### DISCUSSION THREAD

**Discussion by reinhart1010** [tldr-pages/tldr#7480] [Q&A] [answered]
Hi, I'm currently planning to document two other variants of `nvm` command-line utility:

+ **[nvm.fish](https://github.com/jorgebucaran/nvm.fish)** - Specifically designed for the `fish` shell (the original program won't run with it)
+ **[nvm-windows](https://github.com/coreybutler/nvm-windows)** - Specifically designed for Windows

For the case of `nvm-windows` it's easy to document it under `windows/nvm`, but I have some concerns with the `fish` version since it's sharing the same place as `common/nvm` (note that `fish` also runs in BSD and other UNIX-based operating systems, including macOS).

There are some candidates for the document name:

1. `common/nvm.fish` (Unsure
whether this pass the linter)
2. `common/nvm-fish` (Despite `fish` is not a subcommand of the original `nvm`)

**Comment by sbrl** [comment, marked as answer]
Hmm, interesting problem. I'd file it under `common/nvm.fish.md`, and if there's a linting error we should fix that on our end.

**Reply by reinhart1010** [reply]
How about the clients, at least for the official ones? Since we haven't regulate whether file names should include dots between commands. (Who knows that unofficial clients actually do `/^([0-9A-Za-z-]+)./` regex thing on files so `nvm.fish.md` is considered as `nvm` only)

**Reply by reinhart1010** [reply]
I'm opening #7511 and #7512 to add `windows/nvm` and `common/nvm.fish` respectively. Good thing that it passes the linter :smile: .

**Reply by sbrl** [reply]
Hmm, interesting point about the issues with filenames/command names there. We shouldupdate the client spec on that. Would you like to open a pull request about that?

**Reply by EmilyGraceSeville7cf** [reply]
Is it right that we use shell name suffix (`.fish` for instance) whenever some command is designed for non-POSIX compatible shell?

**Reply by reinhart1010** [reply]
Currently there are no specific rule about that. But this could be solved by documenting shell-specific commands into a separate directory, which is also currently being discussed in a recent issue.
